### ISSUE TITLE
Hello, author, thank you very much for your great work. I would like to ask if you can provide the json files of 'infer' and 'Caption', such as '/home/user/selected_face/face_emotion/AU_filter_merge.json' and '/home/user/selected_face/face_emotion/0512_target_smp_end.json'. Thank you very much again.

### ISSUE THREAD

**Issue by Haorotu** [ZebangCheng/Emotion-LLaMA#3]

**Comment by ZebangCheng** [comment]
We have shared the corresponding file links at [MERR](https://drive.google.com/drive/folders/1LSYMq2G-TaLof5xppyXcIuWiSN0ODwqG?usp=drive_link). However, we have renamed the files. **MERR_fine_grained.json** corresponds to **0512_target_smp_end.json**, and **MERR_coarse_grained.json** corresponds to **AU_filter_merge.json**. We apologize for not updating the changes in time.

**Comment by Haorotu** [comment]
Great! Thanks very much again.

**Comment by Haorotu** [comment]
May I ask again, is this Json file generated by code or manually annotated? Thanks

**Comment by Haorotu** [comment]
![running_result](https://github.com/user-attachments/assets/5e1d6869-3070-488b-b2fc-54548347db8d)
Hello, author, I ran 'eval_emotion.py' as required, why did the following result appear? I am confused

**Comment by ZebangCheng** [comment]
> May I ask again, is this Json file generated by code or manually annotated? Thanks

The content in the **MERR_coarse_grained.json** file is generated through a simple combination of model outputs. The **MERR_fine_grained.json** file is filtered by LLaMA3 to select non-repetitive, logically coherent samples. Overall, the content in the JSON files is generated.

**Comment by ZebangCheng** [comment]
> Hello, author, I ran 'eval_emotion.py' as required, why did the following result appear? I am confused

I believe the issue might be due to incorrectly loading the model parameter files. You need to download **Llama-2-7b-chat-hf** and **checkpoint_best.pth** and correctly configure the **eval_emotion.yaml** file. You can refer to the previous issues and responses for detailed steps.

**Comment by Haorotu** [comment]
![runing_result](https://github.com/user-attachments/assets/d42dbf6d-56ad-4195-aadf-916476fd8de7)
![yaml_file](https://github.com/user-attachments/assets/c792c9fd-217c-4424-a538-6cfe993c90c3)
Thank you very much for your reply, it helped me a lot, but the problem still occurs when I import checkpoint.pth, the problem still exists, I am going to crash

**Comment by ZebangCheng** [comment]
Don't worry, we can check the issue step by step. First, ensure that your model parameter files are set to the correct paths. Now, I need to confirm whether you have downloaded the features for different modalities and set the correct paths in the [first_face.py](..//blob/main/minigpt4/datasets/datasets/first_face.py#L161) file as shown below:

```python
def get(self, video_name):
    # FaceMAE feature
    FaceMAE_feats_path = os.path.join(self.file_path, 'mae_340_UTT', video_name + '.npy')
    FaceMAE_feats = torch.tensor(np.load(FaceMAE_feats_path))

    # VideoMAE feature
    VideoMAE_feats_path = os.path.join(self.file_path, 'maeV_399_UTT', video_name + '.npy')
    VideoMAE_feats = torch.tensor(np.load(VideoMAE_feats_path))

    # Audio feature
    Audio_feats_path = os.path.join(self.file_path, 'HL-UTT', video_name + '.npy')
    Audio_feats = torch.tensor(np.load(Audio_feats_path))

    return FaceMAE_feats, VideoMAE_feats, Audio_feats
```

If you want to reproduce the MER2023-SEMI results in `eval_emotion.py`, you need to set the task to **"emotion"** in the [first_face.py](..//blob/main/minigpt4/datasets/datasets/first_face.py#L67) file. This setting restricts the output of Emotion-LLaMA to emotion classification categories in the dataset, such as "happy", "neutral", "sad", "surprise", "worried", "angry".

Please note that the **"Error"** in your image is not a serious issue. It simply checks whether the output result is an emotion category word during testing. The corresponding code is as follows:

```python
for j in range(len(answers)):
    # print("raw answer:", answers[j])
    answers[j] = answers[j].split(" ")[-1]
    targets[j] = targets[j].split(" ")[-1]
    if answers[j] not in ['neutral', 'angry', 'happy', 'sad', 'worried', 'surprise']:
        print("Error: ", answers[j], " Target:", targets[j])
        answers[j] = 'neutral'
```

If your task is set to **"reason"** or **"infer"**, it might cause the output to be incorrect. 

I hope my response helps you. Please feel free to ask if you have any further questions.

**Comment by Haorotu** [comment]
![Configuration](https://github.com/user-attachments/assets/763c7479-3129-4814-8186-4e30780f5920)
Hello, Author, thanks very much for your reply and patient help. There is no problem with the configuration file path from 'first_face.py', the features come from the link given in the previous question, and the task has been set to 'emotion', so I still haven't found the problem, but I noticed that the version of my Python library has been updated. Could this be the reason?

May I ask what the correct output looks like? That is, the output log of eval_emotion.py

**Comment by ZebangCheng** [comment]
If all your settings are correct, your output should not display numerous "Error" messages. The final output for the F1 score on the MER2023-SEMI test set should be around 90%.

I suggest you print the inputs and outputs of Emotion-LLaMA in `eval_emotion.py` for debugging. For example:

```python
for batch in eval_dataloader:
    images = batch['image']
    instruction_input = batch['instruction_input']
    print("instruction_input:", instruction_input)  # Print the model's input instructions
    ...
...
for j in range(len(answers)):
    print("raw answer:", answers[j])  # Print the model's raw output
    ...
```

By doing this, you can better understand what the model is processing and producing, which might help identify any issues.

**Comment by Haorotu** [comment]
Thank you again for your help and reply. I tried to print the output results, taking the "samplenew_00006611.jpg" sample as an example.
![instruction_1](https://github.com/user-attachments/assets/f0e1291d-704e-4e13-b429-86cd721943df)
![instruction_2](https://github.com/user-attachments/assets/bd814bc5-7c53-4ef8-80f0-0bc5cf4ed85a)
‘Instruction’ seems to be fine.
![raw_answer](https://github.com/user-attachments/assets/357d1153-b5ea-442d-9755-f817ea3a8198)
But ‘raw_answer’ has a big problem and does not predict the emotion.
What's going on? Please help me.
